### PR TITLE
fix 8831 - respect --tailcalls- in F# Interactive

### LINKED ISSUE #8831: Setting --tailcall- in FSI commandline options is ignored in VS2019
Whatever combination of FSI startup options I try, none of them seems to have any effect. All I'm trying to do is to see the difference between TCO and non-TCO effects on some functions, but it appears that the settings are ignored.

I tried _Reset Interactive Session_, `#quit` + return, and restarting VS.

image

**Repro steps**

Change any settings by adding [commandline arguments](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/fsharp-interactive-options), reset and observe behavior.

**Expected behavior**

Commandline options should work, after all they're advocated as _"Additional command line arguments passed tot he F# Interactive executable"_. That doesn't appear to happen.

**Actual behavior**

No observable change in FSI.

**Known workarounds**

Probably: run it by hand from the commandline instead of from VS. But that defeats the purpose. Otherwise, none that I know of.

I'm probably missing something obvious here as I'm quite certain that in the past I was able to run it with different settings, but I can't replicate that in any way. In fact, it appears to always run in RELEASE mode, except when I select to enable debugging.

**Related information**

* OS: Windows 7/10
* VS Version 16.4.1 Pro. Also observed in VS2017 Pro

### PR DESCRIPTION
**Pull request by dsyme**
Fix #8831  - we were ignoring the `tailcalls-` flag in ilreflect.fs

I'm not sure we need a specific test case for this - the fix is so simple and I've manually verified by hand that the example in the bug blows up with `--tailcalls-`.   

```
C:\GitHub\dsyme\fsharp> C:\GitHub\dsyme\fsharp\artifacts\bin\fsi\Debug\net472\fsi.exe

Microsoft (R) F# Interactive version 11.0.0.0 for F# 5.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> let map (func: 'a -> 'b) (input: 'a list): 'b list =
-     let rec loop cont =
-         function
-         | [] -> cont []
-         | x::xs -> loop ( fun acc -> cont (func x::acc) ) xs
-
-     loop id input
-
- map id [1..10000000] |> List.last;;
val map : func:('a -> 'b) -> input:'a list -> 'b list
val it : int = 10000000

>
- #q;;

C:\GitHub\dsyme\fsharp> C:\GitHub\dsyme\fsharp\artifacts\bin\fsi\Debug\net472\fsi.exe --tailcalls-

Microsoft (R) F# Interactive version 11.0.0.0 for F# 5.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> let map (func: 'a -> 'b) (input: 'a list): 'b list =
-     let rec loop cont =
-         function
-         | [] -> cont []
-         | x::xs -> loop ( fun acc -> cont (func x::acc) ) xs
-
-     loop id input
-
- map id [1..10000000] |> List.last;;

Process is terminated due to StackOverflowException.
```

### CODE REVIEW

File: fsharplint.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (TIHan) [inline fsharplint.json:1]: I think this file was unintended.

**Review by TIHan — APPROVED** [review]

**Review by cartermp — APPROVED** [review]
will merge once erroneous json file inclusion is dealt with

**Comment by KevinRansom** [comment]
@dsyme just for my education, what would be the scenario when specifying --tailcalls- in fsi be interesting?

**Review by KevinRansom — APPROVED** [review]
The code looks good, I just wonder why fsi ever had the switch for turning off tailcalls, it doesn't seem that useful to me.

**Comment by dsyme** [comment]
> will merge once erroneous json file inclusion is dealt with

Oh yes, I forgot, tbh I think we should just include this file and iterate on it (the settings that are in it in this PR are ok - disabling member names). I can put it in a separate PR though

**Comment by dsyme** [comment]
> The code looks good, I just wonder why fsi ever had the switch for turning off tailcalls, it doesn't seem that useful to me.

It should really get activated when script debugging is used, for alignment with default settings fo debugging project code.  This PR doesn't however do this in VFSI, it just makes the switch work  in FSI

**Comment by KevinRansom** [comment]
@dsyme, script debugging is not a thing on the coreclr.  They have never reimplemented the refemit debugging api's.  But yes that was the one scenario that I could think of too.

**Comment by dsyme** [comment]
I fixed up the merge, this is ready now

> @dsyme, script debugging is not a thing on the coreclr. They have never reimplemented the refemit debugging api's. 

Stack traces are there, just not the source code mappings, so I guess this is still useful today even when running `dotnet fsi` from the command line

**Comment by abelbraaksma** [comment]
Thanks for taking care of this! ❤️

**Pull request merged**